assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import classNames from 'classnames';
+import { parseISO } from 'date-fns';
+import { __ } from '@wordpress/i18n';
+
+import { DisplayStartOrEndDate } from '@sharedServices/filterState';
+import { PLUS_ONE_MONTH, PLUS_TWO_MONTHS } from '@sharedConstants/defaultDates';
+import { BiggieCalendarDate, CalendarDateRange } from '../';
+
+interface Props {
+	className?: string;
+	displayDate: DisplayStartOrEndDate;
+	endDate: string;
+	footerText?: string;
+	headerPrefix?: string;
+	headerText?: string;
+	showDate?: boolean;
+	startDate: string;
+}
+
+const CalendarDateSwitcher: React.FC<Props> = React.memo(
+	({
+		className,
+		displayDate = DisplayStartOrEndDate.start,
+		footerText,
+		headerPrefix,
+		headerText,
+
+		...props
+	}) => {
+		const startDate = parseISO(props.startDate) || PLUS_ONE_MONTH;
+		const endDate = parseISO(props.endDate) || PLUS_TWO_MONTHS;
+
+		switch (displayDate) {
+			case 'end':
+				return (
+					<BiggieCalendarDate
+						className={className}
+						date={endDate}
+						footerText={footerText}
+						headerText={`${headerPrefix || ''}${headerText || __('ends')}`}
+						showTime
+					/>
+				);
+			case 'both':
+				return (
+					<CalendarDateRange
+						className={className}
+						endDate={endDate}
+						footerText={footerText}
+						headerText={headerText}
+						showTime
+						startDate={startDate}
+					/>
+				);
+			case 'start':
+				return (
+					<BiggieCalendarDate
+						className={className}
+						date={startDate}
+						footerText={footerText}
+						headerText={`${headerPrefix || ''}${headerText || __('starts')}`}
+						showTime
+					/>
+				);
+			default:
+				return (
+					<BiggieCalendarDate
+						className={className}
+						date={startDate}
+						footerText={footerText}
+						headerText={`${headerPrefix || ''}${headerText || __('starts')}`}
+					/>
+				);
```

Comment:
maybe just get rid of the `start` case since it is the exact same thing as the `default` case

but make sure that the `default` case gets the `showTime` prop set

Reply:
I think it might be confusing to have `start` option in `DisplayStartOrEndDate`, but not in the switch case.
I have extracted that into a variable and passed for both `default` and `start`, IMHO it's a bit cleaner and safer.